### PR TITLE
#1711 xASL_module_ASL: for 3D spiral 0->NaN

### LINKED ISSUE #1711: Set zeroes outside 3D spiral FoV to NaN
### Description

Torstein Torgan (ASL in CO intoxication, Oslo) had a single GE 3D PCASL scan that was acquired misoriented, see below slice. While most of the cortex is still in the FoV, he noticed that there were some negatives included from interpolation artifacts at the edges, which when divided by in M0 division became extreme large values. These originated from artificial zeroes from the GE 3D spiral FoV which is round, not Cartesian, as the FoV stopped there. Usually this is not a concern because those zeroes are well outside the brain but when they are close to the brain, it is essential that they are set to NaN to be masked out. Like SPM ensures that everything outside the original FoV is set to NaN when moving to a different space.

Henk requested the T1w for this example as well, to check if this influences #1039. It did not.

For Henk to check:
/Users/hjmutsaerts/ExploreASL/TestDataSets/GE_3Dspiral_misaligned_LoQ

image

Now this nicely ends up at the correct location, including the missing FoV:

image

### Tasks

- [x] Very early in the ASL module, set "true" zeroes to NaN (not isnear), for 3D spiral specifically (because with Cartesian the FoV == the reconstruction matrix more or less)
- [x] Add this example to TestDataSets
- [x] Once merged, notify Torstein Torgan

### How to test

- [x] Test this with the new example in TestDataSets

### Release notes

Fix issue with wrongly planned GE ASL scan.

### PR DESCRIPTION
**Pull request by HenkMutsaerts**
### Linked issue

Closes #1711

### How to test

I tested this with https://github.com/ExploreASL/TestDataSets/tree/main/GE_3Dspiral_misaligned_LoQ/
see issue.

### Comments

See issue, please check if you like the current location where I do this. I don't think we should change this for the BIDS conversion, this is an image processing issue.

### CODE REVIEW

File: Modules/xASL_module_ASL.m [hunk 1]
```diff
@@ -213,6 +213,29 @@
 %% G. ASL processing & quantification parameters
 x = xASL_module_ASL_ParseParameters(x, bOutput);
 
+%% H. For GE 3D spiral ASL and M0, check for parts outside of the FoV
+% With a non-Cartesian FoV, there can be voxels in the image matrix that are outside the FoV.
+% If these are not explicitly to NaNs, the original FoV can be wrongly interpreted.
+% This led to a misalignment in the case of registering an ASL where the brain was too close to the edge of the FoV.
+% By setting these voxels to NaNs, the registration algorithms know to skip these voxels outside the FoV.
+
+if strcmpi(x.Q.Vendor, 'GE') && strcmpi(x.Q.MRAcquisitionType, '3D') && strcmpi(x.Q.PulseSequenceType, 'spiral')
```
Comment (jan-petr) [inline Modules/xASL_module_ASL.m]: GE should be regexpi - cause we might have GE_wip.

**Review by jan-petr — CHANGES_REQUESTED** [review]
Minor comment.

File: Modules/xASL_module_ASL.m (hunk 1 above)
Comment (HenkMutsaerts) [reply Modules/xASL_module_ASL.m]: changed to "contains" in b8c3e11495fd6dc8805c5e4aeed3f7d6ed4c1c28

**Review by jan-petr — APPROVED** [review]
OK.

**Pull request merged**